### ISSUE TITLE
3.7.3_Multiplayer Test with AI

### ISSUE THREAD

**Issue by Cyclopia31** [LoneGazebo/Community-Patch-DLL#9966]
_1. Mod Version (X.Y.Z). Current Version: 3.7.3_

3.7.3 Vox Populi EUI

Modpack: https://drive.google.com/file/d/1BZ-SdgZ0FqJzidGO25cgCeuHD5MU2wZE/view?usp=sharing

_2. DirectX Version_

DirectX 11

_3. List of Other Mods_

None

_4. Describe the Issue_

On this new multiplayer stability test I had one CTD from the Host player (maybe just a memory issue), and some desyncs. 
I started to get more and more desyncs after ~Turn 190. I stopped at Turn 209. 
This game is on quick speed and I started at Renaissance Era.
Looking at the out-of-sync value there is maybe something with the trade route range ?

______________________________________________________________________________________
_5. Saves/Logs/CvMinidump

Link of saves, logs and the CvMiniDump file: https://drive.google.com/file/d/1CTTemBGBd4jI4PwJTf-ARja1T5sDYXhb/view?usp=sharing

Thank you for all the work done on multiplayer stability, it gets better with every patch !

**Comment by seroperson** [comment]
Seems like everything started from desynced ethiopian caravel which was on auto-exploration. Possibly auto-exploration logic has some mp-incompatible functions or something like that. I will check it.

**Comment by Cyclopia31** [comment]
Hi, I think I had some force sync even after the caravel died. I can continue the game and post the data if you want ?

**Comment by seroperson** [comment]
No, thank you, I think I have enough data to see what's wrong here. I want to fix this caravel desync (if caravel is really the reason) and after the fix we will go further step-by-step to not accumulate a lot of desync roots in one single game.

**Comment by seroperson** [comment]
Yes, this caravel is the reason. On auto-exploration the logic differs and that's why the caravel is moving differently. Due to it's moving differently, barbs spawn differently, the enemies moves differently, as a result yields can be changed and so on and so on.
![image](https://github.com/LoneGazebo/Community-Patch-DLL/assets/3791221/e7c7194b-d373-46d7-bb8a-48698579f0f7)
So, it moves differently because: 
- there is array of available plots to explore: [CvHomelandAI::GetBestExploreTarget](https://github.com/LoneGazebo/Community-Patch-DLL/blob/4d0be48ec9b3d4089699844234e54fe66411a3de/CvGameCoreDLL_Expansion2/CvHomelandAI.cpp#L222)
- content of this array differs for host and non-host player
- that's why caravels discover different plots on both ends
- this array is changing inside functions `CvEconomicAI::UpdateExplorePlotsFromScratch` and `CvEconomicAI::UpdateExplorePlotsLocally`
- `CvEconomicAI::UpdateExplorePlotsLocally` updates exploration plots iteratively as unit moves
- BUT `CvEconomicAI::UpdateExplorePlotsLocally` is [getting called](https://github.com/LoneGazebo/Community-Patch-DLL/blob/4d0be48ec9b3d4089699844234e54fe66411a3de/CvGameCoreDLL_Expansion2/CvPlot.cpp#L10963C6-L10963C6) just only if `"caravell owner's team" == getActiveTeam()` (`setRevealed` returns `true` in such cases just only if mentioned condition is `true`: [CvPlot::setRevealed](https://github.com/LoneGazebo/Community-Patch-DLL/blob/4d0be48ec9b3d4089699844234e54fe66411a3de/CvGameCoreDLL_Expansion2/CvPlot.cpp#L11546))

tl;dr:
So, this logic makes the exploration on caravell owner's end works like that:
- it moves the unit
- and updates exploration tiles accoring to the revealed tiles

And on the other's ends it works so:
- it moves the unit
- but exploration tiles are not updating according to the revealed tiles

I will prepare a PR. It looks easy as long as `setRevealed`'s return value is used just only in this code block across the whole codebase.
